hi-ml-histopathology/testhisto/testhisto/datamodules/test_slides_datamodule.py

```diff
@@ -24,17 +24,17 @@ def mock_wsi_root_dir(tmpdir_factory: pytest.TempdirFactory) -> py.path.local:
     tmp_root_dir = tmpdir_factory.mktemp("mock_wsi")
     wsi_generator = MockWSIGenerator(
         tmp_path=tmp_root_dir,
-        mock_type=MockWSIType.PATHMNIST,
-        batch_size=1,
-        n_samples=4,
+        mock_type=MockHistoDataType.PATHMNIST,
+        n_tiles=1,
```

Comment:
not entirely sure `batch_size` became `n_tiles`. Shouldn't these be separate parameters?

Reply:
I felt like batch_size is a bit misleading as it not a batch_size for the MockGenerator class itself but rather the batch_size of the dataloader. It's used a bit differently in MockWSI and MockTiles.

It is supposed to define the number of tiles we would like to use to create a Mock WSI. If set to 1 the same tile is repeated n_repeat_tile  times, however if we want a WSI with different tiles we just need to increase it.

As for MockTiles, it determines the number of tiles we would like to have per slide. Speaking of which, we can maybe pick a random number of tiles per slide in the range of n_tiles instead of a fixed number. something like np.random.randint(n_tiles - 5, n_tiles +5), what do you think? 

This is mainly why I renamed the batch_size to n_tiles as it's just the number of tiles we want to sample from the dataloader depending on the type of mock data we want to create.

Reply:
got you, makes sense. `we can maybe pick a random number of tiles per slide in the range of n_tiles ` I think this is a great idea, it would make the mock dataset much closer to the real use case and a number of issues can come from the fact that not all the slides have the same number of tiles so better to test the case